test/unittest/dds/participant/ParticipantTests.cpp

```diff
@@ -2066,6 +2352,77 @@ TEST(ParticipantTests, GetReplierProfileQos)
     ASSERT_EQ(DomainParticipantFactory::get_instance()->delete_participant(participant), RETCODE_OK);
 }
 
+TEST(ParticipantTests, GetReplierQosFromXml)
+{
+    const std::string xml_filename("test_xml_profile.xml");
+    const std::string profile_name("test_replier_profile");
+
+    std::string complete_xml = testing::load_file(xml_filename);
+
+    DomainParticipant* participant =
+            DomainParticipantFactory::get_instance()->create_participant(
+        (uint32_t)GET_PID() % 230, PARTICIPANT_QOS_DEFAULT);
+    ASSERT_NE(participant, nullptr);
+
+    // Get QoS given profile name
+    ReplierQos qos;
+    EXPECT_EQ(
+        participant->get_replier_qos_from_xml(complete_xml, qos, profile_name),
+        RETCODE_OK);
+
+    // Get QoS without providing profile name (gets first one found)
+    ReplierQos qos_empty_profile;
+    EXPECT_EQ(
+        participant->get_replier_qos_from_xml(complete_xml, qos_empty_profile),
+        RETCODE_OK);
+
+    // Check they correspond to the same profile
+    // NOTE: test_replier_profile is assumed to be the first replier profile in the XML file
+    EXPECT_EQ(qos, qos_empty_profile);
+
+    // Load profiles from XML file and get QoS given profile name
+    DomainParticipantFactory::get_instance()->load_XML_profiles_file(xml_filename);
+    ReplierQos qos_from_profile;
+    EXPECT_EQ(
+        participant->get_replier_qos_from_profile(profile_name, qos_from_profile),
+        RETCODE_OK);
+
+    // Check they correspond to the same profile
+    EXPECT_EQ(qos, qos_from_profile);
+
+    // Test return when a non-existent profile is used
+    EXPECT_EQ(
+        participant->get_replier_qos_from_xml(complete_xml, qos, "incorrect_profile_name"),
+        RETCODE_BAD_PARAMETER);
+
+    // Clean up
+    ASSERT_EQ(DomainParticipantFactory::get_instance()->delete_participant(participant), RETCODE_OK);
+}
+
+TEST(ParticipantTests, GetDefaultReplierQosFromXml)
+{
+    const std::string xml_filename("test_xml_profile.xml");
+
+    std::string complete_xml = testing::load_file(xml_filename);
+
+    DomainParticipant* participant =
+            DomainParticipantFactory::get_instance()->create_participant(
+        (uint32_t)GET_PID() % 230, PARTICIPANT_QOS_DEFAULT);
+    ASSERT_NE(participant, nullptr);
+
+    // Get default QoS from XML
+    ReplierQos default_qos;
+    EXPECT_EQ(
+        participant->get_default_replier_qos_from_xml(complete_xml, default_qos),
+        RETCODE_OK);
+
+    // NOTE: cannot load profiles file and compare with default value as
+    // DomainParticipant::get_default_replier_qos is currently unavailable
```

Comment:
But you can use `load_XML_profiles_file` and get the profile `test_replier_profile` which is the default and compare with it.